src/SourceBrowser.Site/Scripts/upload.js

```diff
@@ -0,0 +1,29 @@
+﻿$(document).ready(function () {
+    resetUI();
+    registerButtonClickEvent();
+});
+
+function registerButtonClickEvent() {
+    getButtonFromDom().click(prepareForChange);
+}
+
+function prepareForChange() {
+    getButtonFromDom().prop("disabled", true);
+    getMessageDivFromDom().css("visibility", "visible");
+}
+
+function resetUI() {
+    getButtonFromDom().prop("disabled", false);
+    getMessageDivFromDom().css("visibility", "hidden");
```

Comment:
To prevent stuttering, visibility should be set to hidden immediately, even before the document is ready